test/CadlRanchProjects/inheritance/inheritance.tsp

```diff
@@ -64,6 +66,39 @@ model GoblinShark extends Shark {
   sharktype: "goblin";
 }*/
 
+@doc("extensible enum type for discriminator")
```

Comment:
Do we support
```
@doc("extensible enum type for discriminator")
enum DogKind {
    Golden: "golden",
    Yellow: "yellow"
}

@doc("Test extensible enum type for discriminator")
@discriminator("kind")
model Dog {
    kind: DogKind;
}

model Golden extends Dog {
    kind: DogKind.Golden;
    size: int32;
}

model Yellow extends Dog {
    kind: DogKind.Yellow;
    size: int32;
}
```

Reply:
Yes. What is the difference between your case and the case we have in this PR?

Reply:
Of course they are different. 

This PR:
Golden -> Dog
Cobra -> Snake 

My test:
Golden ->| Dog |
Yellow -->| Dog |

And actually what is the difference between Golden -> Dog and Cobra -> Snake in this PR?

Reply:
> Of course they are different.
> 
> This PR: Golden -> Dog Cobra -> Snake
> 
> My test: Golden ->| Dog | Yellow -->| Dog |
> 
> And actually what is the difference between Golden -> Dog and Cobra -> Snake in this PR?

If you can create one sub-class, for sure you can create multiple sub-classes. There is no need to test two dog case. And Dog is for extensible enum, Snake is for fixed enum.